python/lsst/rubintv/production/redisUtils.py

```diff
@@ -0,0 +1,748 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import json
+import logging
+import os
+import time
+from datetime import timedelta
+
+import redis
+
+from lsst.summit.utils.utils import getSite
+
+from .payloads import Payload
+from .utils import expRecordFromJson
+
+# Check if the environment is a notebook
+clear_output = None
+IN_NOTEBOOK = False
+try:
+    from IPython import get_ipython
+
+    ipython_instance = get_ipython()
+    # Check if notebook and not just IPython terminal. ipython_instance is None
+    # if not in IPython environment, and if IPKernelApp is not in the config
+    # then it's not a notebook, just an ipython terminal.
+    if ipython_instance is None or "IPKernelApp" not in ipython_instance.config:
+        IN_NOTEBOOK = False
+    else:
+        from IPython.display import clear_output
+
+        IN_NOTEBOOK = True
+except (ImportError, NameError):
+    pass
+
+__all__ = "RedisHelper"
+
+
+def decode_string(value):
+    """Decode a string from bytes to UTF-8.
+
+    Parameters
+    ----------
+    value : bytes
+        Bytes value to decode.
+
+    Returns
+    -------
+    str
+        Decoded string.
+    """
+    return value.decode("utf-8")
+
+
+def decode_hash(hash_dict):
+    """Decode a hash dictionary from bytes to UTF-8.
+
+    Parameters
+    ----------
+    hash_dict : dict
+        Dictionary with bytes keys and values.
+
+    Returns
+    -------
+    dict
+        Dictionary with decoded keys and values.
+    """
+    return {k.decode("utf-8"): v.decode("utf-8") for k, v in hash_dict.items()}
+
+
+def decode_list(value_list):
+    """Decode a list of values from bytes to UTF-8.
+
+    Parameters
+    ----------
+    value_list : list
+        List of bytes values to decode.
+
+    Returns
+    -------
+    list
+        List of decoded values.
+    """
+    return [item.decode("utf-8") for item in value_list]
+
+
+def decode_set(value_set):
+    """Decode a set of values from bytes to UTF-8.
+
+    Parameters
+    ----------
+    value_set : set
+        Set of bytes values to decode.
+
+    Returns
+    -------
+    set
+        Set of decoded values.
+    """
+    return {item.decode("utf-8") for item in value_set}
+
+
+def decode_zset(value_zset):
+    """Decode a zset of values from bytes to UTF-8.
+
+    Parameters
+    ----------
+    value_zset : list
+        List of tuple with bytes values and scores to decode.
+
+    Returns
+    -------
+    list
+        List of tuples with decoded values and scores.
+    """
+    return [(item[0].decode("utf-8"), item[1]) for item in value_zset]
+
+
+def getRedisSecret(filename="$HOME/.lsst/redis_secret.ini"):
+    filename = os.path.expandvars(filename)
+    with open(filename) as f:
+        return f.read().strip()
+
+
+def getNewDataQueueName(instrument):
+    return f"INCOMING-{instrument}-raw"
+
+
+class RedisHelper:
```

Comment:
Some documentation of the Redis schema would be really useful: e.g. keys, types, and what they mean.  Absent that, I can't really claim I understood how various states for workers are represented and hence how different methods interact.

Reply:
The short answer is "it's dynamic" 😬 But yes, I take your point - everything needs more documenting I think. Adding it to the list...